### PR TITLE
Update io.fs

### PR DESCRIPTION
**Pull request by Whammo**
Throwing it out there.  Works from 0-30

### CODE REVIEW

**Review by jkotlinski — COMMENTED** [review]
That's real nice thinking, however I guess it will not work when running from cartridge?

**Comment by Whammo** [comment]
That, I did not consider. Isn't Cartridge at $8000?

**Comment by Whammo** [comment]
Works in VICE  w/ cart image.

**Comment by jkotlinski** [comment]
Oh, sorry! I just misremembered from float.fs - but the problem is only in $8000-$9fff range!

File: forth_src/io.fs [hunk 1]
```diff
@@ -53,15 +53,12 @@ w ldx, lsb sta,x
 \ close, and chkin. If ioresult is
 \ nonzero, print error message and
 \ abort.
+
```
Comment (jkotlinski) [inline forth_src/io.fs:56]: was this blank line accidental, or looks better this way?

File: forth_src/io.fs [hunk 2]
```diff
@@ -53,15 +53,12 @@ w ldx, lsb sta,x
...
-8 of ." missing filename" endof
-9 of ." illegal device #" endof
-." io err" endcase cr abort then ;
+?dup if
+rvs 55 1 c! 1-
+1 lshift $a328 + @
+begin dup c@ dup 128 and 0= while
+emit 1+ repeat 128 - emit
+54 1 c!
+cr abort else then ;
```
Comment (jkotlinski) [inline forth_src/io.fs:64]: the `else` looks unneeded

**Review by jkotlinski — CHANGES_REQUESTED** [review]
brilliant idea!

File: forth_src/io.fs (hunk 2 above)
Comment (Whammo) [reply forth_src/io.fs]: artifacts from removing file#

**Comment by Whammo** [comment]
This takes care of the out of range and keeps all the messages available.

```forth
: berr ( ioresult -- )
?dup if
rvs 55 1 c! 1-
2* $a328 + @
begin dup c@ dup 128 and 0= while
emit 1+ repeat 128 - emit
cr abort then ;

\ handle out of range ioresult
: ioabort  ( ioresult -- ? )
dup 9 > if rev ." io err" drop
else berr then ;
```

**Comment by Whammo** [comment]
Credit to Mapping the 64

**Comment by jkotlinski** [comment]
excellent - thanks!

**Comment by jkotlinski** [comment]
hmm, it does not seem to work for me :/

**Comment by Whammo** [comment]
BERR doesn't work?

**Comment by jkotlinski** [comment]
It does not compile, and after fixing compile error, e.g. ”1 IOABORT”
freezes the computer

**Comment by Whammo** [comment]
ugh. Perhaps moving io.fs put it above REV

**Comment by Whammo** [comment]
No, my bad- RVS instead of RSV

**Pull request merged**